st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -401,26 +433,30 @@ def get_one(self, id, exclude_attributes=None, **kwargs):
 
         exclude_fields = self._validate_exclude_fields(exclude_fields=exclude_fields)
 
-        return self._get_one(id=id, exclude_fields=exclude_fields)
+        return self._get_one_by_id(id=id, exclude_fields=exclude_fields,
+                                   requester_user=requester_user,
+                                   permission_type=PermissionType.EXECUTION_VIEW)
 
-    @jsexpose(body_cls=LiveActionCreateAPI, status_code=http_client.CREATED)
-    def post(self, liveaction_api):
-        return self._handle_schedule_execution(liveaction_api=liveaction_api)
+    def post(self, liveaction_api, requester_user=None, **kwargs):
+        return self._handle_schedule_execution(liveaction_api=liveaction_api,
+                                               requester_user=requester_user, **kwargs)
 
-    @request_user_has_resource_db_permission(permission_type=PermissionType.EXECUTION_STOP)
-    @jsexpose(arg_types=[str])
-    def delete(self, exec_id):
+    def delete(self, id, requester_user, **kwargs):
         """
         Stops a single execution.
 
         Handles requests:
             DELETE /executions/<id>
 
         """
-        execution_api = self._get_one(id=exec_id)
+        if not requester_user:
```

Comment:
personally, I don't think we do. Couple extra lines would not break the bank, but this way is much more expressive.